hack/make-rules/update/codegen.sh

```diff
@@ -17,50 +17,66 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-if [[ -n "${BUILD_WORKSPACE_DIRECTORY:-}" ]]; then # Running inside bazel
-  echo "Updating codegen files..." >&2
-elif ! command -v bazel &>/dev/null; then
-  echo "Install bazel at https://bazel.build" >&2
+# darwin is great
+SED=sed
+if which gsed &>/dev/null; then
+  SED=gsed
+fi
+if ! (${SED} --version 2>&1 | grep -q GNU); then
+  echo "!!! GNU sed is required.  If on OS X, use 'brew install gnu-sed'." >&2
   exit 1
-else
-  (
-    set -o xtrace
-    bazel run @io_k8s_test_infra//hack:update-codegen
-  )
-  exit 0
 fi
 
-go_sdk=$PWD/external/go_sdk
-clientgen=$PWD/$1
-deepcopygen=$PWD/$2
-informergen=$PWD/$3
-listergen=$PWD/$4
-go_bindata=$PWD/$5
-controller_gen=$PWD/$6
-do_clean=${7:-}
-
-# Ensure correct go binary is on path
-PATH=$go_sdk/bin:${PATH:-}
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../../.." && pwd -P)"
+cd $REPO_ROOT
+
+echo "Ensuring go version."
+source ./hack/build/setup-go.sh
+
+# build codegen tools
+echo "Install codegen tools."
+cd "hack/tools"
+clientgen=${REPO_ROOT}/_bin/client-gen
+go build -o "${REPO_ROOT}/_bin/client-gen" k8s.io/code-generator/cmd/client-gen
```

Comment:
I think it would make more sense to define make targets for these and make them dependencies of the `update-codegen` target. That would also let us avoid rebuilding them when not needed.

Reply:
Not quite sure that I'm getting the idea. If you mean create a new make target just for installing tools used by update-codegen, it would work but I'm not sure it's the best idea since this new target is used only by update-codegen, maybe not worth the effort of inflating root level Makefile?

For local development, this is pretty fast after the first time, mostly due to go cache I think

Reply:
Ok, sounds good.